### DISCUSSION TITLE
A weird problem after packing to executable

### DISCUSSION THREAD

**Discussion by CanYing0913** [marcelotduarte/cx_Freeze#2499] [Q&A] [answered]
Hi there. I am bumping some of my dependencies to their latest versions so I want to re-freeze my program. However, I now encounter some new errors I forgot how to resolve. I believe I should be able to fix it by modifying my setup script. I am using `python setup.py bdist_msi` on a Windows PC:
![Screenshot 2024-07-14 222515](https://github.com/user-attachments/assets/b6db4e35-2bea-429f-bd78-161801d2a668)
My `setup.py`:
```
from cx_Freeze import setup, Executable

# Dependencies are automatically detected, but they might need fine-tuning
build_exe_options = {
    "build_exe": "dist",
    "includes": [
        "caiman",
    ],
    "packages": [
        "z5py",
    ],
    "include_files": [
        "Fiji.app",
        "config.json"
    ],

}
build_msi_options = {}
build_mac_options = {}
build_appimage_options = {}

executables = [
    Executable(
        "main.py",
        copyright="",
        base="gui",
        # icon="icon.ico",
        target_name="gui",
        shortcut_name="gui",
        shortcut_dir="Program",
    )
]

setup(
    name="name",
    version="0.5",
    description="",
    options={
        "build_exe": build_exe_options,
        "bdist_msi": build_msi_options,
        "bdist_mac": build_mac_options,
        "bdist_appimage": build_appimage_options,
    },
    executables=executables,
)
```

**Comment by marcelotduarte** [comment, marked as answer]
Using conda-forge (or mambaforge) there are issues using some packages.
Try to use cx_Freeze 7.0 or the latest development build:
`pip install --force --no-cache --pre --extra-index-url https://marcelotduarte.github.io/packages/ cx_Freeze`

**Reply by CanYing0913** [reply]
Thanks. Using your dev build fixed it. However, I found some functionalities of my GUI application do not work (and I checked from running on the source code they are working). Are there any ways to debug it? It would be extremely helpful.

**Reply by marcelotduarte** [reply]
If your GUI app is using Qt (PyQt5/6, PySide2/6) you can use some variables to debug in the console, for instance:
set QT_DEBUG=1
set QT_WIN_DEBUG_CONSOLE=1
set QT_LOGGING_RULES='qt.webengine*.debug=true'
set QT_DEBUG_PLUGINS=1
Also, you can use breakpoint() in any part of the code and debug it. Observe that to use debug, base="console" is required.

**Reply by CanYing0913** [reply]
I see. My GUI is based on tkinter actually. I will try to use breakpoint first. Thanks!
